pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -419,10 +438,32 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun, get
 		}
 	}
 
-	if pipelineMeta.VerificationResult != nil && pipelineMeta.VerificationResult.VerificationResultType == trustedresources.VerificationError {
-		err := fmt.Errorf("PipelineRun %s/%s referred pipeline failed signature verification: %w", pr.Namespace, pr.Name, pipelineMeta.VerificationResult.Err)
-		pr.Status.MarkFailed(ReasonResourceVerificationFailed, err.Error())
-		return controller.NewPermanentError(err)
+	if pipelineMeta.VerificationResult != nil {
+		verificationResult := pipelineMeta.VerificationResult
+		switch verificationResult.VerificationResultType {
+		case trustedresources.VerificationError:
+			err := fmt.Errorf("PipelineRun %s/%s referred pipeline failed signature verification: %w", pr.Namespace, pr.Name, verificationResult.Err)
+			pr.Status.MarkFailed(ReasonResourceVerificationFailed, err.Error())
+			pr.Status.SetCondition(&apis.Condition{
+				Type:    trustedresources.ConditionTrustedResourcesVerified,
+				Status:  corev1.ConditionFalse,
+				Message: err.Error(),
+			})
+			return controller.NewPermanentError(err)
+		case trustedresources.VerificationSkip:
+			// do nothing
+		case trustedresources.VerificationWarn:
+			pr.Status.SetCondition(&apis.Condition{
+				Type:    trustedresources.ConditionTrustedResourcesVerified,
+				Status:  corev1.ConditionFalse,
+				Message: verificationResult.Err.Error(),
+			})
+		case trustedresources.VerificationPass:
+			pr.Status.SetCondition(&apis.Condition{
+				Type:   trustedresources.ConditionTrustedResourcesVerified,
+				Status: corev1.ConditionTrue,
+			})
+		}
```

Comment:
The same `switch-case` seems to be repeated. Can this be wrapped into a function to avoid repitition?

Reply:
There are 3 places using the same switch, 2 on pipelinerun and 1 on taskrun. The 2 pipelinerun returns are different. 
Do you have any suggestions? 🤔

Reply:
let me try out, I think we could wrap it into a function

Reply:
The PipelineRuns returns only seem to be happening for the case of `verificationError`. In one case, you return `controller.New...` and in other case you return `nil, controller.New...`. 

What if this function returned `error, conditionToSet`? You could catch that and handle it in both cases?

Reply:
```go
func Verify(verificationResult <typeOfverificationResult>) (error, apis.Condition){
        var condition apis.Condition
        var err error
        
        switch verificationResult.VerificationResultType {
 		case trustedresources.VerificationError:
 			err := fmt.Errorf("PipelineRun %s/%s referred pipeline failed signature    verification: %w", pr.Namespace, pr.Name, verificationResult.Err)
 		 	
// add this line after the output of the function
// pr.Status.MarkFailed(ReasonResourceVerificationFailed, err.Error())
 			condition = &apis.Condition{
 				Type:    trustedresources.ConditionTrustedResourcesVerified,
 				Status:  corev1.ConditionFalse,
 				Message: err.Error(),
 			}
// add this line after the output of the function
 // return controller.NewPermanentError(err)
 		case trustedresources.VerificationSkip:
 			// do nothing
 		case trustedresources.VerificationWarn:
 			condition = &apis.Condition{
 				Type:    trustedresources.ConditionTrustedResourcesVerified,
 				Status:  corev1.ConditionFalse,
 				Message: verificationResult.Err.Error(),
 			}
// set the condition when you catch the output of the function
 		case trustedresources.VerificationPass:
 			condition = apis.Condition{
 				Type:   trustedresources.ConditionTrustedResourcesVerified,
 				Status: corev1.ConditionTrue,
 			}
 		}
 		return err, condition
}
```

Reply:
☝️ is how you would use it well.

Reply:
Does that work?

Reply:
Yeah this should work, just pushed the code.